articles/ai-services/openai/includes/use-your-data-go.md

```diff
@@ -36,91 +36,93 @@ ms.date: 08/29/2023
-           *msgContext.Messages[0].Role,
-           len(*msgContext.Messages[0].Content))
-
-       fmt.Fprintf(os.Stderr, "ChatRole: %s\nChat content: %s\n",
-           *resp.Choices[0].Message.Role,
-           *resp.Choices[0].Message.Content,
-       )
+   
+   	azureOpenAIKey := os.Getenv("AOAIKey")
+   	modelDeploymentID := os.Getenv("AOAIDeploymentId")
+   
+   	// Ex: "https://<your-azure-openai-host>.openai.azure.com"
+   	azureOpenAIEndpoint := os.Getenv("AOAIEndpoint")
+   
+   	// Azure AI Search configuration
+   	searchIndex := os.Getenv("SearchIndex")
+   	searchEndpoint := os.Getenv("SearchEndpoint")
+   	searchAPIKey := os.Getenv("SearchKey")
+   
+   	if azureOpenAIKey == "" || modelDeploymentID == "" || azureOpenAIEndpoint == "" || searchIndex == "" || searchEndpoint == "" || searchAPIKey == "" {
+   		fmt.Fprintf(os.Stderr, "Skipping example, environment variables missing\n")
+   		return
+   	}
+   
+   	keyCredential := azcore.NewKeyCredential(azureOpenAIKey)
+   
+   	// In Azure OpenAI you must deploy a model before you can use it in your client. For more information
+   	// see here: https://learn.microsoft.com/azure/cognitive-services/openai/how-to/create-resource
+   	client, err := azopenai.NewClientWithKeyCredential(azureOpenAIEndpoint, keyCredential, nil)
+   
+   	if err != nil {
+   		//  TODO: Update the following line with your application specific error handling logic
+   		log.Fatalf("ERROR: %s", err)
+   	}
+   	inScope := true
+   	semanticConfiguration := "default"
+   
+   	resp, err := client.GetChatCompletions(context.TODO(), azopenai.ChatCompletionsOptions{
+   		Messages: []azopenai.ChatRequestMessageClassification{
+   			&azopenai.ChatRequestUserMessage{
+   				Content: azopenai.NewChatRequestUserMessageContent("What are the differences between Azure Machine Learning and Azure AI services?"),
+   			},
+   		},
+   		MaxTokens: to.Ptr[int32](512),
+   		AzureExtensionsOptions: []azopenai.AzureChatExtensionConfigurationClassification{
+   			&azopenai.AzureCognitiveSearchChatExtensionConfiguration{
+   				// This allows Azure OpenAI to use an Azure AI Search index.
+   				//
+   				// > Because the model has access to, and can reference specific sources to support its responses, answers are not only based on its pretrained knowledge
+   				// > but also on the latest information available in the designated data source. This grounding data also helps the model avoid generating responses
+   				// > based on outdated or incorrect information.
+   				//
+   				// Quote from here: https://learn.microsoft.com/en-us/azure/ai-services/openai/concepts/use-your-data
+   				Parameters: &azopenai.AzureCognitiveSearchChatExtensionParameters{
+   					Endpoint:  &searchEndpoint,
+   					IndexName: &searchIndex,
+   					Authentication: &azopenai.OnYourDataAPIKeyAuthenticationOptions{
+   						Key: &searchAPIKey,
+   					},
+   					InScope:               &inScope,
```

Comment:
Same question as above do we need these values or can these be dropped to maintain consistency with the engineering sample:

InScope:               &inScope,
SemanticConfiguration: &semanticConfiguration,

https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/ai/azopenai#example-Client.GetChatCompletions-BringYourOwnDataWithCognitiveSearch